modules/dnnlegacy/test/test_caffe_importer.cpp

```diff
@@ -0,0 +1,877 @@
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#include <tuple>
+#include "test_precomp.hpp"
+#include <opencv2/dnnlegacy/dnnlegacy.hpp>
+#include "npy_blob.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+
+using namespace cv;
+
+namespace opencv_test {
+    namespace {
+
+        template<typename Tstring>
+        static std::string _tf(Tstring filename)
+        {
+            return cv::samples::findFile(std::string("dnn/") + filename);
+        }
+
+        class Test_Caffe_nets : public DNNlegacyTestLayer
+        {
+        public:
+            void testFaster(const std::string& proto, const std::string& model, const Mat& ref,
+                            double scoreDiff = 0.0, double iouDiff = 0.0)
+            {
+                checkBackend();
+                cv::dnn::Net net = cv::dnnlegacy::readNetFromCaffe(cv::samples::findFile("dnn/" + proto),
+                                           cv::samples::findFile("dnn/" + model, false));
+                net.setPreferableBackend(backend);
+                net.setPreferableTarget(target);
+
+                if (target == cv::dnn::DNN_TARGET_CPU_FP16)
+                    net.enableWinograd(false);
+
+                Mat img = imread(cv::samples::findFile("dnn/dog416.png"));
+                resize(img, img, Size(800, 600));
+                Mat blob = cv::dnn::blobFromImage(img, 1.0, Size(), Scalar(102.9801, 115.9465, 122.7717), false, false);
+                Mat imInfo = (Mat_<float>(1, 3) << img.rows, img.cols, 1.6f);
+
+                net.setInput(blob, "data");
+                net.setInput(imInfo, "im_info");
+                // Output has shape 1x1xNx7 where N - number of detections.
+                // An every detection is a vector of values [id, classId, confidence, left, top, right, bottom]
+                Mat out = net.forward();
+                scoreDiff = scoreDiff ? scoreDiff : default_l1;
+                iouDiff = iouDiff ? iouDiff : default_lInf;
+                opencv_test::normAssertDetections(ref, out, ("model name: " + model).c_str(), 0.8, scoreDiff, iouDiff);
+            }
+        };
+
+        TEST(Test_Caffe, memory_read)
+        {
+            const std::string proto = cv::samples::findFile("dnn/bvlc_googlenet.prototxt");
```

Comment:
Please add `false` to every large model `findFile` - Let's test build only for now